infra/data_collection/pydantic_models.py

```diff
@@ -373,3 +373,85 @@ class CompleteBenchmarkRun(BaseModel):
     )
     training: Optional[bool] = Field(None, description="ML model benchmarks for training or inference.")
     measurements: List[BenchmarkMeasurement] = Field(description="List of benchmark measurements.")
+
+
+class SweepsResultRecord(BaseModel):
+    """
+    One flattened record for a sweeps test vector execution result.
+
+    This schema is designed to validate the JSON produced by the sweeps runner's
+    file/JSON export and is intentionally decoupled from CI JUnit-based Test.
+    """
+
+    # Identifiers and headers
+    sweep_name: str = Field(description="Name of the sweeps module (e.g., eltwise.unary.relu.relu).")
+    suite_name: Optional[str] = Field(None, description="Suite name within the sweeps module, if applicable.")
+    vector_id: Optional[str] = Field(None, description="Identifier of the specific test vector within the suite.")
+    input_hash: Optional[str] = Field(None, description="Hash of the input vector for deduplication and traceability.")
+
+    # Timing
+    start_time_ts: datetime = Field(description="Timestamp with timezone when the vector execution started.")
+    end_time_ts: datetime = Field(description="Timestamp with timezone when the vector execution ended.")
+
+    # Status & messaging
+    status: Optional[str] = Field(
+        None,
+        description=(
+            "Execution status for this vector. Typical values include: pass, fail_assert_exception, "
+            "fail_l1_out_of_mem, fail_watcher, fail_crash_hang, fail_unsupported_device_perf, skipped, error."
+        ),
+    )
+    message: Optional[str] = Field(None, description="Optional informational message about the execution outcome.")
+    exception: Optional[str] = Field(None, description="Exception text when a failure occurred, if any.")
+    error_signature: Optional[str] = Field(None, description="Derived error signature from the exception, if any.")
+
+    # Performance metrics (optional)
+    e2e_perf: Optional[float] = Field(
+        None, description="End-to-end performance metric for the test (units depend on the specific test)."
+    )
+    device_perf: Optional[Dict[str, Any]] = Field(
```

Comment:
Will you search for a specific key-value pair from this dict field? If so, we might want to normalize this field into a vertical table in the database for faster querying.
You can do something like this:
```suggestion
    device_perf: Optional[set[DevicePerfMetric]] = Field(None, description="A unique set of device-level performance measurements, when device profiling is enabled.")
```

```
class DevicePerfMetric(BaseModel):
    """
    Metric name and its value.
    """
    metric_name: str = Field(description="Output metric name.")
    metric_value_float: Optional[float] = Field(
        description="Output float metric value."
    )
    class Config:
        frozen = True
```

Reply:
This one is tricky because we haven't thoroughly inspected the perf metrics yet for sweeps. Ability to extract performance was implemented in the test runner therefore it was included. However I do like your proposal of creating a Perf class and set of Perf objects for this field. We can include both the e2e_perf and device_perf in this set.